cmake/linux/LinuxDeploy.cmake

```
@@ -0,0 +1,287 @@
+#
+# Copyright (c) 2025, Tres Finocchiaro, <tres.finocchiaro@gmail.com>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+# Variables must be prefixed with "CPACK_" to be visible here
+set(lmms "${CPACK_PROJECT_NAME}")
+set(LMMS "${CPACK_PROJECT_NAME_UCASE}")
+set(ARCH "${CPACK_TARGET_ARCH}")
+set(APP "${CPACK_TEMPORARY_INSTALL_DIRECTORY}/${LMMS}.AppDir")
+
+# Target AppImage file
+set(APPIMAGE_FILE "${CPACK_BINARY_DIR}/${CPACK_PACKAGE_FILE_NAME}.AppImage")
+set(APPIMAGE_BEFORE_RENAME "${CPACK_BINARY_DIR}/${LMMS}-${ARCH}.AppImage")
+
+set(DESKTOP_FILE "${APP}/usr/share/applications/${lmms}.desktop")
+
+# Determine which packaging tool to use
+if(NOT CPACK_TOOL)
+	# Pick up environmental variable
+	if(DEFINED ENV{CPACK_TOOL})
+		set(CPACK_TOOL "$ENV{CPACK_TOOL}")
+	else()
+		set(CPACK_TOOL "appimagetool")
+	endif()
+endif()
+
+# Toggle command echoing & verbosity
+# 0 = no output, 1 = error/warning, 2 = normal, 3 = debug
+if(NOT CPACK_DEBUG)
+	set(VERBOSITY 1)
+	set(APPIMAGETOOL_VERBOSITY "")
+	set(COMMAND_ECHO NONE)
+	set(OUTPUT_QUIET OUTPUT_QUIET)
+else()
+	set(VERBOSITY 2)
+	set(APPIMAGETOOL_VERBOSITY "--verbose")
+	set(COMMAND_ECHO STDOUT)
+	unset(OUTPUT_QUIET)
+endif()
+
+include(DownloadBinary)
+include(CreateSymlink)
+
+# Cleanup CPack "External" json, txt files, old AppImage files
+file(GLOB cleanup "${CPACK_BINARY_DIR}/${lmms}-*.json"
+	"${CPACK_BINARY_DIR}/${lmms}-*.AppImage"
+	"${CPACK_BINARY_DIR}/install_manifest.txt")
+list(SORT cleanup)
+file(REMOVE ${cleanup})
+
+# Download linuxdeploy, expose bundled appimagetool to PATH
+download_binary(LINUXDEPLOY_BIN
+	"https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-${ARCH}.AppImage"
+	linuxdeploy-${ARCH}.AppImage
+	FALSE)
+
+# Symlink nested appimagetool
+set(_APPIMAGETOOL_LINK "${CPACK_CURRENT_BINARY_DIR}/appimagetool")
+if(NOT EXISTS "${_APPIMAGETOOL_LINK}")
+	set(_APPIMAGETOOL "${CPACK_CURRENT_BINARY_DIR}/.linuxdeploy-${ARCH}.AppImage/squashfs-root/plugins/linuxdeploy-plugin-appimage/appimagetool-prefix/AppRun")
+	message(STATUS "Creating a symbolic link ${_APPIMAGETOOL_LINK} which points to ${_APPIMAGETOOL}")
+	create_symlink("${_APPIMAGETOOL}" "${_APPIMAGETOOL_LINK}")
+endif()
+
+# Download linuxdeploy-plugin-qt
+download_binary(LINUXDEPLOY_PLUGIN_BIN
+	"https://github.com/linuxdeploy/linuxdeploy-plugin-qt/releases/download/continuous/linuxdeploy-plugin-qt-${ARCH}.AppImage"
+	linuxdeploy-plugin-qt-${ARCH}.AppImage
+	FALSE)
+
+message(STATUS "Creating AppDir ${APP}...")
+
+file(REMOVE_RECURSE "${CPACK_TEMPORARY_INSTALL_DIRECTORY}/include")
+file(MAKE_DIRECTORY "${APP}/usr")
+
+# Setup AppDir structure (/usr/bin, /usr/lib, /usr/share... etc)
+file(GLOB files "${CPACK_TEMPORARY_INSTALL_DIRECTORY}/*")
+list(SORT files)
+foreach(_file ${files})
+	get_filename_component(_filename "${_file}" NAME)
+	if(NOT _filename MATCHES ".AppDir")
```

```suggestion
	if(NOT _filename MATCHES "\\.AppDir")
```
Since this is regex, I think it needs a double backslash to escape the '.' in CMake